Fix microsite generation, update scala versions, update `sbt-circe-org` plugin version

CJSmith-0141 opened the pull request:
The documentation hasn't been building on Github in a while and there are some features that I would like to document, so this is the first step in fixing the microsite. I also changed two of the badges:

- Removed gitter for Discord
- Changed the maven badge to `0.14.6` instead of the (abandoned?) `0.15.x` branch

CJSmith-0141 committed:
Use CirceOrgSitePlugin for microsite generation

armanbilge commented:
Thanks! You can also run `githubWorkflowGenerate` and delete the docs workflows here.
https://github.com/circe/circe/tree/series/0.14.x/.github/workflows

CJSmith-0141 committed:
bump version of circe for docs to 0.14.6

CJSmith-0141 committed:
make the documentation branch `series/0.14.x`

CJSmith-0141 committed:
runs githubWorkflowGenerate

CJSmith-0141 committed:
deletes old documentation workflows in favor of `CirceOrgSitePlugin`

CJSmith-0141 commented:
> Thanks! 

No worries!

> You can also run `githubWorkflowGenerate` 

done in this commit https://github.com/circe/circe/pull/2218/commits/dbef53943f0f2dc26548d7bbbd33b17a003244b5

> and delete the docs workflows here. https://github.com/circe/circe/tree/series/0.14.x/.github/workflows

done in this commit https://github.com/circe/circe/pull/2218/commits/a821bd08b05756395748f4955a8f506ba7b3a551

Thank you for the quick feedback @armanbilge :+1:

CJSmith-0141 committed:
update docs dependencies to versions that exist

CJSmith-0141 committed:
make directory.conf more idiomatic

CJSmith-0141 committed:
Bumps version of scala, sbt-circe-org

- Scala `2.13`    -> `2.13.12`
- Scala `3`       -> `3.3.1`
- `sbt-circe-org` -> `0.3.0`
- runs `githubWorkflowGenerate`

CJSmith-0141 commented:
Ok after pushing an update to https://github.com/circe/sbt-circe-org to get new laika functionality, I had to update to the latest versions of scala 2.13 and scala 3. Hope that's ok! 

https://github.com/circe/circe/pull/2218/commits/820e0d7ee499ffe4759917957e749fb2bc9099a3

CJSmith-0141 committed:
revert scala 3 update (doesn't compile)

zarthross commented on docs/codecs/known-issues.md:
```diff
@@ -39,9 +35,8 @@ title:  "Warnings and known issues"
-   problems, it's likely that they're not your fault. Please file an issue here or ask a question on
-   the [Gitter channel](https://gitter.im/circe/circe), and we'll do our best to figure out whether the problem is
-   something we can fix.
+   problems, it's likely that they're not your fault. Please file an issue here or ask a question on [Discord](https://discord.gg/XF3CXcMzqD), 
+   and we'll do our best to figure out whether the problem is something we can fix.
```
Can we link to, or reference the specific channel within the typelevel discord?

CJSmith-0141 replied:
I'll give it a shot, sure.

zarthross committed:
Merge branch 'series/0.14.x' into series/0.14.x

CJSmith-0141 committed:
Addresses @zarthross comment on #2218

- https://github.com/circe/circe/pull/2218#discussion_r1450992632
- Links to #circe channel on typelevel discord

CJSmith-0141 committed:
Merge branch 'series/0.14.x' of https://github.com/CJSmith-0141/circe into series/0.14.x

adamnfish commented on .github/workflows/ci.yml:
```diff
@@ -356,43 +393,91 @@ jobs:
+  site:
+    name: Generate Site
+    strategy:
+      matrix:
```
Given this matrix, could the steps below be streamlined? There are checks for various Java versions, but it looks like they're unneccesary if we've picked `temurin@11` for this job?

CJSmith-0141 replied:
I tried to change `githubWorkflowJavaVersions` for just the `docs` project and run `sbt githubWorkflowGenerate` and that didn't produce a diff.

armanbilge replied:
> There are checks for various Java versions, but it looks like they're unneccesary if we've picked `temurin@11` for this job?

@adamnfish this is something that we could fix in sbt-typelevel. The reason there are checks for all the Java versions is because the setup steps for the docs is inherited from the setup steps for the primary CI jobs which _do_ need all those Java versions. This inheritance is useful because in some cases the setup does other things (e.g. install native dependencies or setup a database/server) that need to be shared across the CI and docs jobs.

CJSmith-0141 committed:
remove `%{ include ... }%` from docs

CJSmith-0141 committed:
Merge branch 'series/0.14.x' into series/0.14.x

hamnis committed:
Merge branch 'series/0.14.x' into series/0.14.x

hamnis commented on project/plugins.sbt:
```diff
@@ -2,4 +2,4 @@ addSbtPlugin("org.portable-scala" % "sbt-scala-native-crossproject" % "1.3.2")
 addSbtPlugin("org.scala-native" % "sbt-scala-native" % "0.4.17")
 addSbtPlugin("pl.project13.scala" % "sbt-jmh" % "0.4.6")
-addSbtPlugin("io.circe" % "sbt-circe-org" % "0.1.4")
+addSbtPlugin("io.circe" % "sbt-circe-org" % "0.3.0")
```
```suggestion
addSbtPlugin("io.circe" % "sbt-circe-org" % "0.3.1")
```

hamnis committed:
upgrade sbt-circe-org to 0.3.1

hamnis committed:
regenerate github workflow

hamnis reviewed (approved):
Great work. Sorry for the very long delay
